protocol/da-footprint-block-limit.md

```diff
@@ -0,0 +1,299 @@
+This can lead to network congestion for batchers if the throughput on L2s surpasses that of L1 blob space, or if blob
+space is congested for other reasons.
+
+Current L1 throughput is 6 blobs per block at target with a max of 9 blobs per block.
+This is approx. `128 kB * 6 / 12 s = 64 kB/s` or `96 kB/s`, resp. The L1 cost is proportional to the L1 origin’s base
+fee and blob base fee. But Most OP Stack chains have gas targets high enough to accept far more than 128kb of calldata
+per (2s) block. So the calldata floor cost doesn't nearly limit calldata throughput enough to prevent.
+
+The current measure to counter high DA throughput that's being deployed on productio networks is batcher throttling.
+When a network is throttled at the builder policy-level, however, its base fee can dive and lead to priority fee
+auctions and other unwanted user experience issues.
+
+Also see the design doc [Custom Calldata Floor Gas](https://github.com/ethereum-optimism/design-docs/pull/294) for a
+more detailed analysis of the same problem and context and a similar solution.
+
+## Proposed Solution
+
+A _DA footprint block limit_ is introduced to limit the total amount of estimated compressed transaction data that can
+fit into a block. The base fee update rules take this new total DA footprint into account, so the base fee market isn't
+broken like for policy-level throttling solutions.
+
+The main idea is to introduce a new _DA footprint_ resource that is tracked next to gas, and limit it by the same
+block gas limit. This is inspired by the
+[Multidimensional Gas Metering](https://ethresear.ch/t/a-practical-proposal-for-multidimensional-gas-metering/22668)
+design, but the regular gas isn't touched, like in this design for L1 Ethereum.
+
+Similarly to the regular gas usage of transaction calldata, we calculate a transaction’s _DA footprint_ by taking
+its RLP-encoding, calculate the compressed transaction DA usage estimation from
+[Fjord](https://specs.optimism.io/protocol/fjord/exec-engine.html#fjord-l1-cost-fee-changes-fastlz-estimator)
+and multiply it by a factor, the _DA footprint gas scalar_, to get to the transaction’s DA footprint. 
+
+```python
+da_usage_estimate = max(minTransactionSize, intercept + fastlzCoef*fastlzSize / 1e6)
+da_footprint = da_usage_estimate * da_footprint_gas_scalar
+```
+
+Here, `fastlzSize` is the length of the FastLZ-compressed RLP-encoding of a transaction.
+
+Now when evaluating whether a transactions still fits into a block, we take the maximum over the two resources, regular
+total gas use of all transactions, and new total DA footprint, and see if this max is still below the block gas limit.
+The Multidimensional Gas Metering design introduces a new block field `gas_metered` to differentiate it from `gas_used`
+as the sum total of all transaction's gas used. However, it is proposed to just repurpose a block's `gas_used` field to
+hold the maximum over both resources' totals:
```

Comment:
I would expect `gas_used` header fields that don't match the sum of the gas used by each transaction to break something somewhere. I don't know what would break, but I'd be surprised if nothing is relying on that invariant.